cmd/ci-operator/main.go

```diff
@@ -519,12 +523,18 @@ func (o *options) Complete() error {
 
 	o.clusterConfig = clusterConfig
 
-	if len(o.pullSecretPath) > 0 {
-		o.pullSecret, err = getPullSecretFromFile(o.pullSecretPath)
+	if o.pullSecretPath != "" {
+		o.pullSecret, err = getSecret(o.pullSecretPath)
 		if err != nil {
 			return fmt.Errorf("could not get pull secret from path %s: %w", o.pullSecretPath, err)
 		}
 	}
+	if o.pushSecretPath != "" {
```

Comment:
This is the missing logic in the reverted PR.
`ci-operator` is broken in non-promotion jobs because the secret is mounted only for promotion jobs.